sorting/count_inversions.cpp

```diff
@@ -0,0 +1,267 @@
+/**
+ * @file
+ * @brief Counting Inversions using [Merge
+ Sort](https://en.wikipedia.org/wiki/Merge_sort)
+ *
+ * @details
+ * Program to count the number of inversions in an array
+ * using merge-sort.
+ *
+ * The count of inversions help to determine how close the array
+ * is to being sorted in ASCENDING order.
+ *
+ * two elements a[i] and a[j] form an inversion if a[i] > a[j] and i < j
+ *
+ * Time Complexity --> O(n.log n)
+ * Space Complexity --> O(n) ; additional arrat temp[1..n]
+ * ### Algorithm
+
+ *   1. The idea is similar to merge sort, divide the array into two equal or
+ almost
+ *      equal halves in each step until the base case is reached.
+ *   2. Create a function merge that counts the number of inversions when two
+ halves of
+ *      the array are merged, create two indices i and j, i is the index for
+ first half
+ *      and j is an index of the second half. if a[i] is greater than a[j], then
```

Comment:
When using Doxygen try to stick to MarkDown rules. The visibility of these a[i] is not great as you can see from the picture:
<img width="1503" alt="image" src="https://user-images.githubusercontent.com/21105489/100096417-e5f2f400-2e53-11eb-98b2-fe17462fdeb7.png">

whereas using the \`a[i]\` makes it easy to read which you can see from this (I changed one of your comments in gitpod to show this):
<img width="610" alt="Screenshot 2020-11-24 at 12 53 17" src="https://user-images.githubusercontent.com/21105489/100096516-0a4ed080-2e54-11eb-9551-36faea7c6cd1.png">


```suggestion
 *      and j is an index of the second half. if `a[i]` is greater than `a[j]`, then
```